app/controllers/users_controller.rb

```diff
@@ -1,7 +1,14 @@
 class UsersController < ApplicationController
+  # The index action is responsible for displaying a list of users. It fetches all user records from the database
+  # assigns them to an instance variable @users.
+  # This instance variable can be accessed within the corresponding view file to display the list of users.
   def index
```

Comment:
- [ ] I encountered an issue when creating a `user` from the rails console and after debugging for a couple of minutes I noticed the issue was from the validation to the `posts_counter` you applied keep in mind a `user` can be created without specifying the `posts_counter` value so please make sure to fix this (you can give it an intial value of 0). :+1: :100: 

```ruby
validates :posts_counter, numericality: { greater_than_or_equal_to: 0, allow_nil: true }
```